`make push-multiarch-mimir` prints lot of warnings

Running `make push-multiarch-mimir` prints many warnings like:

```
make push-multiarch-mimir

make GOOS=linux GOARCH=amd64 BINARY_SUFFIX=_linux_amd64 cmd/mimir/mimir
make[1]: Entering directory '/Users/peter/Grafana/mimir'
./docs/docs.mk:34: warning: overriding recipe for target 'docs-docker-rm'
docs/docs.mk:34: warning: ignoring old recipe for target 'docs-docker-rm'
./docs/docs.mk:38: warning: overriding recipe for target 'docs-pull'
docs/docs.mk:38: warning: ignoring old recipe for target 'docs-pull'
./docs/docs.mk:43: warning: overriding recipe for target 'docs'
docs/docs.mk:43: warning: ignoring old recipe for target 'docs'
Makefile:7: warning: overriding recipe for target 'help'
./Makefile:7: warning: ignoring old recipe for target 'help'
Makefile:72: warning: overriding recipe for target 'image-tag'
./Makefile:72: warning: ignoring old recipe for target 'image-tag'
Makefile:103: warning: overriding recipe for target 'push-multiarch-mimir'
./Makefile:103: warning: ignoring old recipe for target 'push-multiarch-mimir'
Makefile:111: warning: overriding recipe for target 'fetch-build-image'
./Makefile:111: warning: ignoring old recipe for target 'fetch-build-image'
Makefile:118: warning: overriding recipe for target 'push-multiarch-build-image'
./Makefile:118: warning: ignoring old recipe for target 'push-multiarch-build-image'
Makefile:138: warning: overriding recipe for target 'images'
./Makefile:138: warning: ignoring old recipe for target 'images'
Makefile:208: warning: overriding recipe for target 'exes'
./Makefile:208: warning: ignoring old recipe for target 'exes'
Makefile:208: warning: overriding recipe for target 'cmd/metaconvert/metaconvert'
./Makefile:208: warning: ignoring old recipe for target 'cmd/metaconvert/metaconvert'
Makefile:208: warning: overriding recipe for target 'cmd/mimir/mimir'
...
```

These warnings seem to have been introduced by commit 667ac5d1c5. It would be nice to get rid of them.

--- a/Makefile
+++ b/Makefile
@@ -130,7 +130,7 @@ push-multiarch-build-image:
 # 'make: Entering directory '/go/src/github.com/grafana/mimir' phase.
 DONT_FIND := -name vendor -prune -o -name .git -prune -o -name .cache -prune -o -name .pkg -prune -o -name mimir-mixin-tools -prune -o -name trafficdump -prune -o
 
-MAKEFILES = $(shell find . $(DONT_FIND) \( -name 'Makefile' -o -name '*.mk' \) -print)
+MAKE_FILES = $(shell find . $(DONT_FIND) \( -name 'Makefile' -o -name '*.mk' \) -print)
 
 # Get a list of directories containing Dockerfiles
 DOCKERFILES := $(shell find . $(DONT_FIND) -type f -name 'Dockerfile' -print)
@@ -382,11 +382,11 @@ endif
 
 .PHONY: check-makefiles
 check-makefiles: format-makefiles
-	@git diff --exit-code -- $(MAKEFILES) || (echo "Please format Makefiles by running 'make format-makefiles'" && false)
+	@git diff --exit-code -- $(MAKE_FILES) || (echo "Please format Makefiles by running 'make format-makefiles'" && false)
 
 .PHONY: format-makefiles
 format-makefiles: ## Format all Makefiles.
-format-makefiles: $(MAKEFILES)
+format-makefiles: $(MAKE_FILES)
 	$(SED) -i -e 's/^\(\t*\)  /\1\t/g' -e 's/^\(\t*\) /\1/' -- $?
 
 clean:

--- a/Makefile
+++ b/Makefile
@@ -130,6 +130,9 @@ push-multiarch-build-image:
 # 'make: Entering directory '/go/src/github.com/grafana/mimir' phase.
 DONT_FIND := -name vendor -prune -o -name .git -prune -o -name .cache -prune -o -name .pkg -prune -o -name mimir-mixin-tools -prune -o -name trafficdump -prune -o
 
+# MAKE_FILES is a list of make files to lint.
+# We purposefully avoid MAKEFILES as a variable name as it influences
+# the files included in recursive invocations of make
 MAKE_FILES = $(shell find . $(DONT_FIND) \( -name 'Makefile' -o -name '*.mk' \) -print)
 
 # Get a list of directories containing Dockerfiles

